core/shared/platform/linux-sgx/sgx_wamr.edl

```diff
@@ -118,19 +118,34 @@ enclave {
         int ocall_pthread_rwlock_unlock([user_check]void *rwlock);
 
         int ocall_get_errno();
-        int ocall_socket(int domain, int type, int protocol);
+
+        /* sockets */
+        int ocall_accept(int sockfd, [in, size=addr_size]void *addr,
+                  [in, out, size=4] uint32_t *addrlen, uint32_t addr_size);
+        int ocall_bind(int sockfd, [in, size=addrlen]const void *addr,
+                 uint32_t addrlen);
+        int ocall_connect(int sockfd, [in, out, size=addrlen]void *addr, uint32_t addrlen);
```

Comment:
Will the content in addr be written back for enclave to use? If not, how about changing to `[in, size=addrlen]`? As in Linux, addr is const in connect:
```C
int connect(int sockfd, const struct sockaddr *addr,
                   socklen_t addrlen);
```

Reply:
You are correct! After reading the docs, the `connect` function does not alter the socket structure. I realized it was also true for the function `accept`, where I removed the `out` annotation as well.